devito/finite_differences/differentiable.py

```diff
@@ -219,3 +309,23 @@ class Mod(sympy.Mod, Differentiable):
 evalf_table[Add] = evalf_table[sympy.Add]
 evalf_table[Mul] = evalf_table[sympy.Mul]
 evalf_table[Pow] = evalf_table[sympy.Pow]
+
+
+_args_sortkey = cmp_to_key(sympy.Basic.compare)
+
+
+def _addsort(args):
+    # in-place sorting of args
+    args.sort(key=_args_sortkey)
+
+
+def to_differentiable(expr):
```

Comment:
I'm sure I'll complain about this, when I'll review it

Reply:
I know you will. but I'm not rewriting solve